File: chapter04/src/main/java/com/ixiongyu/interfaces/javaversion/Button.java
package com.ixiongyu.interfaces.javaversion;

/**
 * @author xiongyu
 * @version Create at ：2020/8/23 下午8:46
 */

public class Button implements Clickable, Focusable {
    @Override
    public void showOff() {
        Clickable.super.showOff();
        Focusable.super.showOff();
    }

    @Override
    public void click() {

    }

    @Override
    public void setFocus(Boolean focus) {
    }
}

File: chapter04/src/main/java/com/ixiongyu/interfaces/Button.kt
package com.ixiongyu.interfaces

/**
 * @author  xiongyu
 * @version Create at ：2020/8/23 下午8:34
 */
open class Button : Clickable, Focusable {
    override fun click() {
        println("click  me")
    }

    override fun showOff() {
        super<Clickable>.showOff()
        super<Focusable>.showOff()
    }


}
